packages/build-tools/src/generic.ts

```diff
@@ -18,19 +17,22 @@ export async function runGenericJobAsync(ctx: BuildContext<Generic.Job>): Promis
 
   await addEasWorkflows(customBuildCtx);
 
-  const relativeConfigPath = nullthrows(
-    ctx.job.customBuildConfig?.path,
-    'Missing job definition file path.'
-  );
-  const configPath = path.join(customBuildCtx.projectSourceDirectory, relativeConfigPath);
-
   const globalContext = new BuildStepGlobalContext(customBuildCtx, false);
 
-  const parser = new BuildConfigParser(globalContext, {
-    externalFunctions: getEasFunctions(customBuildCtx),
-    externalFunctionGroups: getEasFunctionGroups(customBuildCtx),
-    configPath,
-  });
+  const parser = ctx.job.steps
+    ? new StepsConfigParser(globalContext, {
+        externalFunctions: getEasFunctions(customBuildCtx),
+        externalFunctionGroups: getEasFunctionGroups(customBuildCtx),
+        steps: ctx.job.steps,
+      })
+    : new BuildConfigParser(globalContext, {
+        externalFunctions: getEasFunctions(customBuildCtx),
+        externalFunctionGroups: getEasFunctionGroups(customBuildCtx),
+        configPath: path.join(
+          customBuildCtx.projectSourceDirectory,
+          ctx.job.customBuildConfig.path
```

Comment:
Should this `nullthrows`?

Reply:
I don't think so it's needed here 🤔 

Based on the `Generic.Job` type (and validation) we know that either `ctx.job.customBuildConfig.path` or `ctx.job.steps`